src/rdkafka_queue.c

```diff
@@ -587,6 +614,10 @@ int rd_kafka_q_serve_rkmessages (rd_kafka_q_t *rkq, int timeout_ms,
                                        RD_KAFKA_Q_CB_RETURN, NULL);
                 if (res == RD_KAFKA_OP_RES_KEEP ||
                     res == RD_KAFKA_OP_RES_HANDLED) {
```

Comment:
The rko will have been destroyed if res == HANDLED, and the line below will thus dereference freed memory.